mne/epochs.py

```diff
@@ -2147,3 +2147,45 @@ def add_channels_epochs(epochs_list, name='Unknown', add_eeg_ref=True,
     epochs._projector, epochs.info = setup_proj(epochs.info, add_eeg_ref,
                                                 activate=proj)
     return epochs
+
+
+def _compare_epochs_infos(info1, info2, ind):
+    """Compare infos"""
+    if not info1['nchan'] == info2['nchan']:
+        raise ValueError('epochs[%d][\'info\'][\'nchan\'] must match' % ind)
+    if not info1['bads'] == info2['bads']:
+        raise ValueError('epochs[%d][\'info\'][\'bads\'] must match' % ind)
+    if not info1['sfreq'] == info2['sfreq']:
+        raise ValueError('epochs[%d][\'info\'][\'sfreq\'] must match' % ind)
+    if not set(info1['ch_names']) == set(info2['ch_names']):
+        raise ValueError('epochs[%d][\'info\'][\'ch_names\'] must match' % ind)
+    if len(info2['projs']) != len(info1['projs']):
+        raise ValueError('SSP projectors in epochs files must be the same')
+    if not all(proj_equal(p1, p2) for p1, p2 in
+               zip(info2['projs'], info1['projs'])):
+        raise ValueError('SSP projectors in epochs files must be the same')
+
+
+def concatenate_epochs(epochs_list):
+    """Concatenate a list of epochs into one epochs object
+
+    Parameters
+    ----------
+    raws : list
+        list of Epochs instances to concatenate (in order).
+
+    Returns
+    -------
+    epochs : instance of Epochs
+        The result of the concatenation (first Epochs instance passed in).
+    """
+    out = epochs_list[0].copy()
+    out._data = out.get_data()
+    out.preload = True
+    for ii, epochs in enumerate(epochs_list[1:]):
+        _compare_epochs_infos(epochs.info, epochs_list[0].info, ii)
+        if not np.all(epochs.times == epochs_list[0].times):
+            raise ValueError('Epochs must have same times')
+
+        out._data = np.concatenate((out._data, epochs.get_data()), axis=0)
```

Comment:
I would construct the EpochsArray at the end and avoid the hack of accessing epochs._data which is meant to be private.